Mathlib/GroupTheory/Subgroup/Basic.lean

```diff
@@ -2076,6 +2076,19 @@ theorem center_toSubmonoid : (center G).toSubmonoid = Submonoid.center G :=
 #align subgroup.center_to_submonoid Subgroup.center_toSubmonoid
 #align add_subgroup.center_to_add_submonoid AddSubgroup.center_toAddSubmonoid
 
+/-- For a group with zero, the center of the units is the same as the units of the center. -/
+def centerUnitsEquivUnitsCenter (G₀ : Type*) [GroupWithZero G₀] :
```

Comment:
```suggestion
@[simps]
def centerUnitsEquivUnitsCenter (G₀ : Type*) [GroupWithZero G₀] :
```
or similar, again checking the lemmas look ok.

Reply:
I had removed this earlier, because I didn't particularly like the options it gave me before. I tried `simps!` (which I've never seen before) and it seems to have gone a bit too far:

```lean
[simps.verbose] adding projection Subgroup.val_centerUnitsEquivUnitsCenter_apply_coe:
      ∀ (G₀ : Type u_5) [inst : GroupWithZero G₀] (a : { x // x ∈ Subgroup.center G₀ˣ }),
        ↑↑(↑(Subgroup.centerUnitsEquivUnitsCenter G₀) a) = ↑↑a 

[simps.verbose] adding projection Subgroup.val_inv_centerUnitsEquivUnitsCenter_apply_coe:
      ∀ (G₀ : Type u_5) [inst : GroupWithZero G₀] (a : { x // x ∈ Subgroup.center G₀ˣ }),
        ↑↑(↑(Subgroup.centerUnitsEquivUnitsCenter G₀) a)⁻¹ = (↑↑a)⁻¹ 

[simps.verbose] adding projection Subgroup.val_centerUnitsEquivUnitsCenter_symm_apply_coe:
      ∀ (G₀ : Type u_5) [inst : GroupWithZero G₀] (u : { x // x ∈ Submonoid.center G₀ }ˣ),
        ↑↑(↑(MulEquiv.symm (Subgroup.centerUnitsEquivUnitsCenter G₀)) u) = ↑↑u 

[simps.verbose] adding projection Subgroup.val_inv_centerUnitsEquivUnitsCenter_symm_apply_coe:
      ∀ (G₀ : Type u_5) [inst : GroupWithZero G₀] (u : { x // x ∈ Submonoid.center G₀ }ˣ),
        ↑(↑(↑(MulEquiv.symm (Subgroup.centerUnitsEquivUnitsCenter G₀)) u))⁻¹ = ↑↑u⁻¹ 
```

So I will add the first and the third manually

Reply:
actually, screw it, I'll just use `simps!` as long as Lean doesn't complain

Reply:
You can use `@[simps val_apply_coe val_symm_apply_coe]` to request it only generate those ones (or some similar syntax)

Reply:
I guess that doesn't work so well with the `to_additive (attr := simps!)`?

Reply:
ah, it does, ends up being `apply_coe_val`